### PR TITLE
Remove Backwards Compatible Tag Methods

### PR DESCRIPTION
**Pull request by damcclean**
This pull request removes references to tag methods that were moved across for backwards compatibility sake when building Statamic v3. Addresses #1932 

Here is a list of the changes I've made:
* I've removed the references to `$this->get`, `$this->getBool`, `$this->getFloat`, $this->getInt`, `$this->getParam` and `$this->getList`, referencing methods on the [ArrayAccessor](https://github.com/statamic/cms/blob/master/src/Tags/ArrayAccessor.php) instead (where the $this->params stuff comes from)
* I've removed the `HasParameters` trait from the two places it was being used (inside `Tags` and `RendersForms`
* Also removed the actual `HasParameters` trait from the codebase.
* I've also removed the `parameters` property that was being defined on the `Tags` class, also for backwards compatibility reasons. Again, I had to update some references to that with the new `params` property.

I noticed that the `Entry` tag and the `Terms` tag have their own `parameters` property, presumably because they don't extend upon the `Tags` class, like all other tags. Would you like me to also rename those properties?

Let me know if you have any questions.

### CODE REVIEW

File: src/Tags/Assets.php [hunk 1]
```diff
@@ -132,9 +132,9 @@ private function sort()
      */
     private function limit()
     {
-        $limit = $this->getInt('limit');
+        $limit = $this->params->get('limit');
         $limit = ($limit == 0) ? $this->assets->count() : $limit;
-        $offset = $this->getInt('offset');
+        $offset = $this->params->get('offset');
```
Comment (jasonvarga) [inline src/Tags/Assets.php:137]: Should be `int()`

File: src/Tags/Concerns/RendersForms.php [hunk 2]
```diff
@@ -24,7 +23,7 @@ protected function formOpen($action, $method = 'POST', $knownTagParams = [])
             'action' => $action,
         ];
 
-        if ($this->getBool('files')) {
+        if ($this->params->get('files')) {
```
Comment (jasonvarga) [inline src/Tags/Concerns/RendersForms.php:26]: Should be `bool()`

File: src/Tags/GetFiles.php [hunk 3]
```diff
@@ -38,8 +38,8 @@ public function index()
      */
     private function getFiles()
     {
-        $folders = $this->get(['in', 'from']);
-        $depth = $this->getInt('depth', 1);
+        $folders = $this->params->get(['in', 'from']);
+        $depth = $this->params->get('depth', 1);
```
Comment (jasonvarga) [inline src/Tags/GetFiles.php:42]: Should be `int()`

File: src/Tags/GetFiles.php [hunk 4]
```diff
@@ -121,34 +121,34 @@ private function filterExtension()
...
         }
     }
 
     private function limit()
     {
-        $limit = $this->getInt('limit');
+        $limit = $this->params->get('limit');
         $limit = ($limit == 0) ? $this->files->count() : $limit;
-        $offset = $this->getInt('offset');
+        $offset = $this->params->get('offset');
```
Comment (jasonvarga) [inline src/Tags/GetFiles.php:144]: Should both be `int()`

File: src/Tags/Glide.php [hunk 5]
```diff
@@ -140,8 +140,8 @@ private function output($url)
                 compact('url', 'width', 'height')
             );
         }
-        if ($this->getBool('tag')) {
-            return "<img src=\"$url\" alt=\"{$this->get('alt')}\" />";
+        if ($this->params->get('tag')) {
```
Comment (jasonvarga) [inline src/Tags/Glide.php:143]: Should be `bool()`

File: src/Tags/Glide.php [hunk 6]
```diff
@@ -163,7 +163,7 @@ private function generateGlideUrl($item)
             return;
         }
 
-        $url = ($this->getBool('absolute')) ? URL::makeAbsolute($url) : URL::makeRelative($url);
+        $url = ($this->params->get('absolute')) ? URL::makeAbsolute($url) : URL::makeRelative($url);
```
Comment (jasonvarga) [inline src/Tags/Glide.php:166]: Should be `bool()`

File: src/Tags/Locales.php [hunk 7]
```diff
@@ -129,12 +129,12 @@ private function getData()
...
     private function sort($locales)
     {
-        if ($sort = $this->get('sort')) {
+        if ($sort = $this->params->get('sort')) {
             [$sort, $dir] = $this->getSort($sort);
             $locales = ($dir === 'asc') ? $locales->sortBy($sort) : $locales->sortByDesc($sort);
         }
 
-        if ($this->getBool('current_first', true)) {
+        if ($this->params->get('current_first', true)) {
```
Comment (jasonvarga) [inline src/Tags/Locales.php:137]: Should be `bool()`

File: src/Tags/Path.php [hunk 8]
```diff
@@ -16,13 +16,13 @@ class Path extends Tags
...
         // If no src param was used, we will treat this as a regular `path` variable.
-        if (! $src = $this->get(['src', 'to'])) {
+        if (! $src = $this->params->get(['src', 'to'])) {
             return array_get($this->context, 'path');
         }
 
         $url = PathAPI::tidy(Config::getSiteUrl().$src);
 
-        if ($this->getBool('absolute', false)) {
+        if ($this->params->get('absolute', false)) {
```
Comment (jasonvarga) [inline src/Tags/Path.php:25]: Should be `bool()`

File: src/Tags/Theme.php [hunk 9]
```diff
@@ -53,15 +53,15 @@ public function asset()
...
 
         $path = 'img/'.$src;
 
         $url = $this->themeUrl($path);
 
-        $alt = $this->get('alt');
+        $alt = $this->params->get('alt');
 
-        if ($this->getBool('tag')) {
+        if ($this->params->get('tag')) {
```
Comment (jasonvarga) [inline src/Tags/Theme.php:64]: Should be `bool()`

File: src/Tags/Theme.php [hunk 10]
```diff
@@ -75,13 +75,13 @@ public function img()
...
     {
-        $src = $this->get('src', 'app');
+        $src = $this->params->get('src', 'app');
 
         $path = 'js/'.Str::ensureRight($src, '.js');
 
         $url = $this->themeUrl($path);
 
-        if ($this->getBool('tag')) {
+        if ($this->params->get('tag')) {
```
Comment (jasonvarga) [inline src/Tags/Theme.php:84]: Should be `bool()`

File: src/Tags/Theme.php [hunk 11]
```diff
@@ -95,13 +95,13 @@ public function js()
...
     {
-        $src = $this->get('src', 'app');
+        $src = $this->params->get('src', 'app');
 
         $path = 'css/'.Str::ensureRight($src, '.css');
 
         $url = $this->themeUrl($path);
 
-        if ($this->getBool('tag')) {
+        if ($this->params->get('tag')) {
```
Comment (jasonvarga) [inline src/Tags/Theme.php:104]: Should be `bool()`

File: src/Tags/Theme.php [hunk 12]
```diff
@@ -129,31 +129,31 @@ public function output()
...
+            return [$this->params->get('as', 'output_contents') => $contents];
         }
 
         return $contents;
     }
 
     private function themeUrl($path)
     {
-        if ($this->getBool('version')) {
+        if ($this->params->get('version')) {
```
Comment (jasonvarga) [inline src/Tags/Theme.php:140]: Should be `bool()`

File: src/Tags/Theme.php [hunk 13]
```diff
@@ -129,31 +129,31 @@ public function output()
...
         );
 
-        if ($this->getBool('cache_bust')) {
+        if ($this->params->get('cache_bust')) {
             throw_if(! File::exists($path = public_path($path)), new \Exception("File $path does not exist."));
             $url .= '?v='.File::lastModified($path);
         }
 
-        if (! $this->getBool('absolute')) {
+        if (! $this->params->get('absolute')) {
```
Comment (jasonvarga) [inline src/Tags/Theme.php:156]: Should both be `bool()`

File: src/Tags/TransChoice.php [hunk 14]
```diff
@@ -11,9 +11,9 @@ class TransChoice extends Tags
      */
     public function wildcard($tag)
     {
-        $key = $this->get('key', $tag);
-        $count = $this->getInt('count', 1);
+        $key = $this->params->get('key', $tag);
+        $count = $this->params->get('count', 1);
```
Comment (jasonvarga) [inline src/Tags/TransChoice.php:15]: Should be `int()`

**Comment by jasonvarga** [comment]
This is wonderful, thank you!

I haven't done an in-depth review yet, but have given you some initial things to address.

**Comment by damcclean** [comment]
> This is wonderful, thank you!
> 
> I haven't done an in-depth review yet, but have given you some initial things to address.

No problem, saw it was on the v3 final project and wanted to help a hand.

I've made your requested changes, not sure why I didn't catch them yesterday 😂

**Comment by jesseleite** [comment]
Thanks @damcclean!  Missed a few things, but will commit those separately.

**Comment by jesseleite** [comment]
Thanks @damcclean!!  Missed a few things but addressed those separately (https://github.com/statamic/cms/commit/9abfb9aa7fc84ff20fa4eb5e438952973f00909c).

**Pull request merged**